dotcom-rendering/scripts/deploy/riff-raff.yaml

```diff
@@ -3,21 +3,33 @@ regions: [eu-west-1]
 allowedStages:
   - CODE
   - PROD
+templates:
+  cloudformation:
+    type: cloud-formation
+    parameters:
+      amiEncrypted: true
+      amiTags:
+        # Keep the Node version in sync with `.nvmrc`
+        Recipe: dotcom-rendering-ARM-jammy-node-18.17.0
+        AmigoStage: PROD
 deployments:
   frontend-cfn:
-    type: cloud-formation
+    template: cloudformation
     parameters:
       templateStagePaths:
         CODE: DotcomRendering-CODE.template.json
         PROD: DotcomRendering-PROD.template.json
       cloudFormationStackByTags: false
       cloudFormationStackName: rendering
       amiParameter: AMIRendering
-      amiEncrypted: true
-      amiTags:
-        # Keep the Node version in sync with `.nvmrc`
-        Recipe: dotcom-rendering-ARM-jammy-node-18.17.0
-        AmigoStage: PROD
+  render-front:
+    template: cloudformation
+    parameters:
+      templateStagePaths:
+        CODE: DotcomRendering-front-CODE.template.json
+      cloudFormationStackByTags: false
+      cloudFormationStackName: render-front
+      amiParameter: AMIRenderfront
```

Comment:
how does this work?

Reply:
@JamieB-gu do you have more insight on this - I can't really see where it's used?

Reply:
When you run `cdk synth` (in this project `make cdk-synth`) you'll see it in the generated cloudformation template.

AFAIK (@jacobwinch can correct me if I've got any of this wrong) AMIgo[^1] can be used to define an AMI with particular features, known as a recipe, and "bake" (build) it on a regular schedule. RiffRaff uses `amiTags` defined in the `riff-raff.yaml` file to derive the latest version of an AMI that's baked to a particular AMIgo recipe, and gets an id for this image. It passes this id into cloudformation during deployment[^2] via a parameter, and this is the name of that parameter. Cloudformation needs to know what name to expect and what it means (image id), so this will also be defined in the cloudformation template.

In our case we generate the cloudformation template using CDK, which automatically sets the name of this parameter to `"AMI" + "name of your app"`[^3]. Previously we only had one app, `rendering`, so this parameter was always `AMIRendering`. This PR creates a new app called `render-front`, so now there's a second version of this parameter called `AMIRenderfront`.

This PR sets up RiffRaff to deploy both apps together, so we have two sets of deployments defined:

- One for the rendering app, using the `DotcomRendering` cloudformation template containing the `AMIRendering` parameter.
- One for the render-fronts app, using the `DotcomRendering-front` cloudformation template containing the `AMIRenderfronts` parameter.

AFAIK @jamesgorrie you would like this new fronts app to use the same AMIgo recipe as the existing rendering app. Therefore we're able to use a shared deployment template[^4][^5] to define most of the AMI configuration (like `amiTags`), only diverging when we hit this `amiParameter` field because it's different for each app as mentioned. As we always want the same recipe, I think it's preferable to only define and update it in one place, which also makes it a bit easier to run the `check-node-versions` script:

https://github.com/guardian/dotcom-rendering/blob/3a1ef2e597f36f9bc531138f36535eb71b2447e0/scripts/check-node-versions.mjs#L37-L38

[^1]: https://github.com/guardian/amigo
[^2]: https://github.com/guardian/amigo/blob/5edc0ff7daaaee94f3697c35f2729b1a12391510/docs/riffraff-integration.md
[^3]: https://github.com/guardian/cdk/blob/402a5bb57f696461cd1c4f7afc2220fc525d3700/docs/migration-guide-ec2.md?plain=1#L191
[^4]: https://github.com/guardian/riff-raff/blob/main/riff-raff/public/docs/reference/riff-raff.yaml.md#templates
[^5]: https://github.com/guardian/riff-raff/blob/main/riff-raff/public/docs/reference/riff-raff.yaml.md#template

Reply:
This all sounds correct to me @JamieB-gu!